api/src/main/java/org/openmrs/PatientProgramAttribute.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs;
+
+import org.openmrs.attribute.Attribute;
+import org.openmrs.attribute.BaseAttribute;
+
+public class PatientProgramAttribute extends BaseAttribute<ProgramAttributeType, PatientProgram> implements Attribute<ProgramAttributeType, PatientProgram> {
+    private Integer patientProgramAttributeId;
+
+    @Override
+    public Integer getId() {
+        return getPatientProgramAttributeId();
+    }
+
+    @Override
+    public void setId(Integer id) {
+        setPatientProgramAttributeId(id);
+    }
+
+    public PatientProgram getPatientProgram() {
+        return getOwner();
+    }
+
+    public void setPatientProgram(PatientProgram patientProgram) {
+        setOwner(patientProgram);
```

Comment:
You seem to be doing the above differently from the original source. Was that by mistake?

Reply:
no it wasn't the bahmni source has a bahmniPatientProgram class whose functionality I ported over to the PatientProgram class, so I'm just returning an instance of the patientProgram class since it has all the needed functionality

Reply:
Did you notice that the original source constructs a new object?

Reply:
Also the constructor for PatientProgram expects a `patientProgramId`

Reply:
Yes, which then calls super in the bahmniPatienProgram class which then calls super on the patientProgramAttribute class. let me rewrite the logic